### DISCUSSION TITLE
Micro-guides for basic CMS features

### DISCUSSION THREAD

**Discussion by jamespohalloran** [tinacms/tinacms#2667] [Ideas]
A few of us internally were thinking it would be nice to have a "Guides" section of the docs dedicated to typical CMS-features that might not be out-of-the-box Tina features per say, but are still doable with a custom workflow / setup.

One example of a micro-guide that was brought up was "Drafts". 
This can be achieved in a few ways:
- with a branching workflow
- by adding a "draft" toggle field to a collection's fields list (The page renderer would be responsible for conditionally rendering it).

A few more examples of topics:
- i18n
- SEO
- Editorial workflow
- Content Scheduling 
- "Editing Theme/Layout-level data with Tina" https://github.com/tinacms/tinacms/issues/2503
- Colocated schema + component
- Using storybook components with Tina

Even if a feature isn't optimally supported, the guide could mention that, and it would at least give the docs-reader a baseline to work with. 

- Would this type of thing be useful?
- Any other common CMS features that we should consider adding to the above list?

**Comment by scottgallant** [comment]
Great idea. Would we want to open up other frameworks to these guys once supported? I.e. Tina + Hugo, or Tina + Remix?

**Comment by scottgallant** [comment]
Micro guide suggestions: 

* Colocated schema + component 
* Using storybook components with Tina

**Comment by solidevolution** [comment]
Setup multilange site for i18n is unclear.
Is there a concept for translated urls? Is it possible to create one page for multiple languages?

**Reply by perkinsjr** [reply]
Hi there,

Currently Tina doesn't directly support this, you would need to implement your own version using next.js supported i8n. 

For example a Discord user implemented there own version using Filenames.

```
type ParseTinaFileNameResponseType = {
  slug: string
  locale?: string
}

/**
 * @description Takes in a Tina filename with the locale appended (e.g. "some-blogpost.en")
 * @param {String} filename
 * @returns {ParseTinaFileNameResponseType}
 */
const parseTinaFileName = (
  filename: string | undefined
): ParseTinaFileNameResponseType => {
  if (typeof filename !== 'string') {
    return {
      slug: '',
    }
  }

  const parts = filename.split('.')

  if (parts.length < 2) {
    return {
      slug: filename,
    }
  }

  const slug = parts[0]
  const locale = parts.length > 1 ? parts[1] : undefined

  return {
    slug,
    locale,
  }
}

export default parseTinaFileName
```


```

Then handle the rest from there.
